feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go

```diff
@@ -0,0 +1,1946 @@
+			WithIndex(1).WithIPAddress(gribiIPv4EntryDefVRF1),
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(2).WithIPAddress(gribiIPv4EntryDefVRF2),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(1).AddNextHop(1, 1).AddNextHop(2, 3).WithBackupNHG(18),
+		fluent.IPv4Entry().WithNetworkInstance(niTeVrf111).
+			WithPrefix(gribiIPv4EntryVRF1111+"/"+maskLen32).WithNextHopGroup(1),
+
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(3).WithIPAddress(gribiIPv4EntryDefVRF4),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(4).AddNextHop(3, 2).WithBackupNHG(19),
+		fluent.IPv4Entry().WithNetworkInstance(niTeVrf111).
+			WithPrefix(gribiIPv4EntryVRF1112+"/"+maskLen32).WithNextHopGroup(4),
+	)
+	if err := awaitTimeout(args.ctx, t, args.client, time.Minute); err != nil {
+		t.Logf("Could not program entries via client, got err, check error codes: %v", err)
+	}
+
+	teVRF111IPList := []string{gribiIPv4EntryVRF1111, gribiIPv4EntryVRF1112}
+	for ip := range teVRF111IPList {
+		chk.HasResult(t, args.client.Results(t),
+			fluent.OperationResult().
+				WithIPv4Operation(teVRF111IPList[ip]+"/32").
+				WithOperationType(constants.Add).
+				WithProgrammingResult(fluent.InstalledInFIB).
+				AsResult(),
+			chk.IgnoreOperationID(),
+		)
+	}
+
+	// Programming AFT entries for prefixes in ENCAP_TE_VRF_A
+	args.client.Modify().AddEntry(t,
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(201).WithEncapsulateHeader(fluent.IPinIP).
+			WithIPinIP(ipv4OuterSrc111, gribiIPv4EntryVRF1111).
+			WithNextHopNetworkInstance(niTeVrf111),
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(202).WithEncapsulateHeader(fluent.IPinIP).
+			WithIPinIP(ipv4OuterSrc111, gribiIPv4EntryVRF1112).
+			WithNextHopNetworkInstance(niTeVrf111),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(11).AddNextHop(201, 1).AddNextHop(202, 3).WithBackupNHG(18),
+		fluent.IPv4Entry().WithNetworkInstance(niEncapTeVrfA).
+			WithPrefix(gribiIPv4EntryEncapVRF+"/"+maskLen24).WithNextHopGroup(11),
+		fluent.IPv6Entry().WithNetworkInstance(niEncapTeVrfA).
+			WithPrefix(gribiIPv6EntryEncapVRF+"/"+maskLen126).WithNextHopGroup(11),
+	)
+
+	if err := awaitTimeout(args.ctx, t, args.client, time.Minute); err != nil {
+		t.Logf("Could not program entries via client, got err, check error codes: %v", err)
+	}
+
+	chk.HasResult(t, args.client.Results(t),
+		fluent.OperationResult().
+			WithIPv4Operation(gribiIPv4EntryEncapVRF+"/24").
+			WithOperationType(constants.Add).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			AsResult(),
+		chk.IgnoreOperationID(),
+	)
+
+	// Install an 0/0 static route in ENCAP_VRF_A and ENCAP_VRF_B pointing to the DEFAULT VRF.
```

Comment:
Should this be an actual static route instead of gRIBI route for the Encap VRF fallback ?
Although OC local routing does'nt specify a leaf NH VRF leaf in the model , there is a VRF fallback leaf in OC NI model.

Reply:
Its as per README

* Install an 0/0 static route in ENCAP_VRF_A and ENCAP_VRF_B pointing to the DEFAULT VRF.

Reply:
Yes, but its not a Static route, its a gRIBI programmed route here.